apis/v1beta1/httproute_types.go

```diff
@@ -263,6 +263,56 @@ type HTTPRouteRule struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms, as specified in GEP-2257,
+// and MUST BE >= 1ms or "0s" to disable (no timeout).
+//
+// +kubebuilder:validation:XValidation:message="backendRequest timeout cannot be longer than request timeout",rule="!(has(self.request) && has(self.backendRequest) && duration(self.request) != duration('0s') && duration(self.backendRequest) > duration(self.request))"
+type HTTPRouteTimeouts struct {
+	// Request specifies the duration for processing an HTTP client request after which the
```

Comment:
Potentially a can of worms here, but seeing `Request` right beside `BackendRequest` feels a bit funny here. I know "Request Timeout" is a very common term so understand how we got there, and I think that both `Timeouts.Request` and `Timeouts.BackendRequest` independently make sense, but they feel a bit confusing when combined. Would any of the following pairs be clearer?

1. `ClientRequest` and `BackendRequest`
2. `FullRequest` and `BackendRequest`
3. `Request` and `PerTry`

This does not need to block, just would like to get a bit more feedback/discussion on potential name pairs before merging this.

/cc @frankbu @kflynn

Reply:
This is really a GEP1742 discussion, which we've already had: https://github.com/kubernetes-sigs/gateway-api/pull/1997#discussion_r1211967593

That said, I still prefer it as is, but could live with `ClientRequest` and `BackendRequest`.

Reply:
Thanks for the link @frankbu! I'd assumed this was already covered in the GEP, just wanted to make sure we'd considered the alternatives here. I agree with what you and @youngnick said in that thread that "Request Timeout" is a well-known and broadly used term, and any of the alternatives for that feel inferior. I also can't come up with a great alternative to "Backend Request Timeout", so I'll just mark this as resolved.